### ISSUE TITLE
Homey cannot find Switchbot?

### ISSUE THREAD

**Issue by SortOfSmart** [AdyRock/com.switchbot#12]
I've tried adding Switchbot to my Homey a couple of times now, did a factory reset or three, put the little bot literally next to my Homey, but I simply cannot get it to work. I also cannot use the Switchbot app itself for some reason, it simply won't sign me up so I can't make an account there. 

Did I understand it wrong? I thought the Switchbot Hub wasn't needed when paired with Homey. I may still need it in the future because I'm not sure what the range is, but for now I just want to try and see it work.

**Comment by SortOfSmart** [comment]
I've got it! :-)

So yeah, I did select BLE, and the thing had been lying beside my Homey for two or three days, so that wasn't the problem. I did come across another post here that suggested rebooting the Homey and yup, there it was. Homey can control the bot now. He's such a cute little guy :-)

I'm assuming with the hub, you'll be able to see in Homey if the command came through? I don't like to have to guess if it actually worked or not.

**Comment by AdyRock** [comment]
What firmware are you running on Homey?
Athom made a few improvements to BLE from v5 onwards. I'm on v7 and it works well.

Depending on the Bot mode, you should see the current state in the app, e.g. if it is **not** set to a momentary press, then even if you change the state in the SwitchBot phone app, it should show the new state in Homey (after about 30 seconds).

The main benefit of the hub is you can place them closer to the devices they control, so if Homey is too far away then it doesn't matter.
Also, you can control the Bots through the SwitchBot phone app even when you are not in range.
The disadvantages are: it requires an internet connection, the more devices you add the slower the feedback and not all features are supported (battery level, etc).

If you are happy to play with tech, then I also have a ESP32 app that you can program into a cheap ESP32 module. This converts the BLE data to WiFi for the Homey app and is very fast to respond (1 or 2 seconds) and you can have multiple modules around the house to provide total BLE coverage.
The code for that is in my GitHub space so you need to but a module (Amazon sell then for around £8), download the code and setup your WiFi information in a file, install the code on the module with the Arduino program and then the Homey app automatically detects them.

**Comment by SortOfSmart** [comment]
V7 here as well. I'll have to look into the app settings. I like to control my devices through the Homey app and/or my Google home assistant as much as possible, so don't use native apps a lot. Actually for this specific project I don't need to be able to control it remotely. In fact, I'm trying to set it up so I cannot trigger it by accident. That's simple to do with a flow, but I haven't found a way to prevent me from accidentally pressing the device when I'm just scrolling though my devices though. Homey is next to the button the Bot should press, so I'm still doubtful the hub is actually needed.

As for the rest.. sorry, I'm really a beginner at all this. My technical skill started and ended when I made a pressure mat that controlled a bicycle lamp with a 4,5 volt battery and some paperclips when I as ten. I can install a smart switch behind an existing light switch, and that's about it :-) So I'll just have to buy the hub if I want more options.

**Comment by AdyRock** [comment]
I agree about the accidental press and I am on the fence as to whether the quick action is a good feature or not. Personally I think Athom would have done better to make the 'quick action' a long press and the open device as a tap. At least then the accidental presses would be less problematic as if the details are display you can just backup. However, too late to change that now.
I might look to see if the quick action can be turned on and off for built in capabilities such as the on/off button.

**Comment by SortOfSmart** [comment]
That would've been great. Even better if they would add the possibility to choose which action you want if you tap or hold. Or even be able to set it on a device by device basis. I'd also settle for a way to hide a certain device so it doesn't show up on the main screen, so it can _only_ be triggered through a flow.

I've already moved the bot to a non-existent room where I have all my non-switchable devices such as door sensors, on the bottom of the screen. But of course, that still shows on the home screen. I mean.. we all have devices we would rather not trigger by accident, right?
